hamlpy/parser/filters.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import unicode_literals
+
+import sys
+import textwrap
+
+# Required on Python 2 to accept non-unicode output
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+# Pygments and Markdown are optional dependencies which may or may not be available
+try:
+    import pygments
+    from pygments.formatters import HtmlFormatter
+    from pygments.lexers import guess_lexer, PythonLexer
+    from pygments.util import ClassNotFound
+
+    _pygments_available = True
+except ImportError:  # pragma: no cover
+    _pygments_available = False
+
+try:
+    from markdown import markdown as markdown_lib
+
+    _markdown_available = True
+except ImportError:  # pragma: no cover
+    _markdown_available = False
+
+
+from future.utils import raise_from
+
+from .generic import ParseException
+
+
+def plain(content, indent, options):
+    return textwrap.dedent(content)
+
+
+def cdata(content, indent, options):
+    return indent + '<![CDATA[\n' \
+           + content + '\n' \
+           + indent + ']]>'
+
+
+def css(content, indent, options):
+    return '<style type=%(attr_wrapper)stext/css%(attr_wrapper)s>\n' \
+           '/*<![CDATA[*/\n' % {'attr_wrapper': options['attr_wrapper']} \
+           + content + '\n' \
+           + '/*]]>*/\n</style>'
+
+
+def stylus(content, indent, options):
+    return indent + '<style type=%(attr_wrapper)stext/stylus%(attr_wrapper)s>\n' \
```

Comment:
There are some inconsistencies with now indentation is handled by different filters. I've tried to preserve some of those inconsistencies for now so that the template tests work as is